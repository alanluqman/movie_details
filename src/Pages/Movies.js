/* eslint-disable react/jsx-one-expression-per-line */
import axios from 'axios';
import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { fetchMovies } from '../Redux/movies';
import '../style/Movies.css';

function Movies() {
  const movies = useSelector((state) => state.movies);
  const dispatch = useDispatch();
  const api = 'https://api.tvmaze.com/shows';
  useEffect(() => async () => {
    if (movies.length === 0) {
      const movieList = await axios.get(api).then((Response) => Response.data);
      dispatch(fetchMovies(movieList));
    }
  }, []);

  return (
    <div className="movieListHolder">
      <div className="header">
        <h1>Movie List</h1>
        <input type={Text} className="search-box" placeholder="Search..." />
      </div>
      <ul className="moviesDisplay">
        {
        movies.map((movie) => (
          <li key={movie.id}>
            <Link to={`/${movie.name}`}>
              <img className="poster" src={movie.image.medium} alt="poster" />
              <p className="movie-name">{movie.name}</p>
            </Link>
          </li>
        ))
        }
      </ul>
      {/* <Link to={`/${movie.name}`}>{movie.name}</Link> */}
    </div>
  );
}

export default Movies;
