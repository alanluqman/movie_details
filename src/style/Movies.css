.movieListHolder {
  max-width: 1000px;
  margin: auto;
}

.header {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
}

.header h1 {
  font-family: Verdana, Geneva, Tahoma, sans-serif;
}

.search-box {
  padding: 10px;
  border: 0;
  border-radius: 10px;
  background-color: rgb(63, 63, 63);
  color: white;
}

.search-box:focus {
  outline: 0;
}

ul {
  margin: 0;
}

li {
  list-style-type: none;
}

.moviesDisplay {
  display: grid;
  grid-template-columns: 1fr 1fr;
  padding: 10px;
  gap: 10px;
}

.poster {
  width: 100%;
}

.movie-name {
  font-family: 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
  color: rgb(255, 185, 79);
}

@media screen and (min-width: 400px) {
  .header {
    flex-direction: row;
  }
}

@media screen and (min-width: 500px) {
  .moviesDisplay {
    grid-template-columns: repeat(3, 1fr);
  }
}

@media screen and (min-width: 900px) {
  .moviesDisplay {
    grid-template-columns: repeat(4, 1fr);
  }
}
