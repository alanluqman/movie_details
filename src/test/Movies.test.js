import {
  render, screen, act,
} from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../Redux/configStore';
import Movies from '../Pages/Movies';

describe('Movies page component', () => {
  afterEach(() => {
    act(() => store.dispatch({
      type: 'moviesdetails/movies/getMovies',
      payload: [],
    }));
  });
  // check for Movie List title render
  it('check for Movie List title render', async () => {
    render(
      <Provider store={store}>
        <Movies />
      </Provider>,
    );
    expect(screen.getByText('Movie List')).toMatchSnapshot();
  });
  // check for search box render
  it('check for search box render', async () => {
    render(
      <Provider store={store}>
        <Movies />
      </Provider>,
    );
    expect(screen.getByPlaceholderText('Search...')).toMatchSnapshot();
  });
});
