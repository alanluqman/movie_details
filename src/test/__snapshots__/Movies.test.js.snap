// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies page component check for Movie List title render 1`] = `
<h1>
  Movie List
</h1>
`;

exports[`Movies page component check for search box render 1`] = `
<input
  class="search-box"
  placeholder="Search..."
/>
`;
